.gitlab-ci.yml

```diff
@@ -0,0 +1,107 @@
+
+stages:
+  - lint
+  - test
+  - build
+
+variables:                         &default-vars
+  GIT_STRATEGY:                    fetch
+  GIT_DEPTH:                       100
+  CARGO_INCREMENTAL:               0
+  CARGO_TARGET_DIR:                "/ci-cache/${CI_PROJECT_NAME}/targets/${CI_COMMIT_REF_NAME}/${CI_JOB_NAME}"
+  CI_IMAGE:                        "paritytech/ci-linux:production"
+
+.rust-info-script:                 &rust-info-script
+  - rustup show
+  - cargo --version
+  - rustup +nightly show
+  - cargo +nightly --version
+  - bash --version
+  - sccache -s
+
+.docker-env:                       &docker-env
+  image:                           "${CI_IMAGE}"
+  before_script:
+    - *rust-info-script
+    - ./scripts/ci/pre_cache.sh
+    - sccache -s
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+      - unknown_failure
+      - api_failure
+  interruptible:                   true
+  tags:
+    - linux-docker
+
+.kubernetes-env:                   &kubernetes-env
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+      - unknown_failure
+      - api_failure
+  interruptible:                   true
+  tags:
+    - kubernetes-parity-build
+
+.collect-artifacts:                &collect-artifacts
+  artifacts:
+    name:                          "${CI_JOB_NAME}_${CI_COMMIT_REF_NAME}"
+    when:                          on_success
+    expire_in:                     28 days
+    paths:
+      - ./artifacts/
+
+#### stage:                        lint
+
+fmt:
+  stage:                           lint
+  image:                           "${CI_IMAGE}"
+  <<:                              *kubernetes-env
+  script:
+    - cargo +nightly fmt --all -- --check --config merge_imports=true
+
+clippy:
+  stage:                           lint
+  <<:                              *docker-env
+  script:
+    # add clippy is temporary until tests in substrate are not fixed
+    # https://github.com/paritytech/substrate/pull/9637
+    - rustup component add clippy --toolchain nightly-x86_64-unknown-linux-gnu
+    - cargo +nightly clippy --all-targets -- -D warnings
+
+#### stage:                       test
+
+deny:
```

Comment:
In previous config deny was also running periodically (nightly) to make sure changes are detected before a PR is created. Obviously it only makes sense with email notifications to detect the failures. If it's cumbersome to implement, we can keep it as-is.

Reply:
I think it's doable but currently it's still unclear for me whether I need to finish this PR or just close it since the repo is going to merged in Substrate project.

Reply:
Yeah, fair enough, let's skip it.